pkg/restore/restore.go

```diff
@@ -2527,3 +2539,36 @@ func (ctx *restoreContext) handleSkippedPVHasRetainPolicy(
 	obj = resetVolumeBindingInfo(obj)
 	return obj, nil
 }
+
+func hasDisabledPVReprovisioningStorageClass(unstructuredPV *unstructured.Unstructured, ctx *restoreContext) (bool, error) {
+	disabledStorageClasses := ctx.restore.Spec.DisabledPVReprovisioningStorageClasses
+
+	// Converting Unstructured to PV object.
+	pv := new(v1.PersistentVolume)
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredPV.Object, pv); err != nil {
+		return false, errors.Wrapf(err, "error converting persistent volume to structured")
+	}
+
+	// Checking if PV StorageClass is in the DisabledPVReprovisioningStorageClasses list.
+	for _, disabledStorageClass := range disabledStorageClasses {
+		if disabledStorageClass == pv.Spec.StorageClassName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (ctx *restoreContext) handleSkippedPVHasDisabledReprovisioningStorageClass(
+	obj *unstructured.Unstructured,
+	logger logrus.FieldLogger,
+) (*unstructured.Unstructured, error) {
+	logger.Infof("Restoring persistent volume as-is because it doesn't have a snapshot and it's storage class has re-provisionning disabled.")
```

Comment:
```suggestion
	logger.Infof("Restoring persistent volume as-is because it doesn't have a snapshot and restore has storage class re-provisionning disabled.")
```